app/components/channel_list/categories/header/index.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import withObservables from '@nozbe/with-observables';
+
+import CategoryHeader from './header';
+
+import type CategoryModel from '@typings/database/models/servers/category';
+
+const withCategory = withObservables(['category'], ({category}: {category: CategoryModel}) => ({
+    category,
+    hasChannels: category.hasChannels,
```

Comment:
Why handle this here instead of in the parent component, or even the component itself? (0/5 on where it should be handled)

Reply:
The value comes through as an observable when passed through here if I understand correctly (so `hasChannels`, when it switches between true/false, actually re-renders any components dependant on it).

Reply:
But the parent component receives a category, that if has changed, will render the parent. Then the parent can pass the `hasChannels`, which will re-render the component, right?

But also, if the child is already receiving the category, it will re-render itself when the category changes, so... it can check the hasChannels itself on its own render function.

(I may be missing something, so feel free to let me know if what I am saying is wrong)

Reply:
I think it's more because of the `@lazy` decorator being used to create `hasChannels` on the model; when called it will return the value at that moment (not observed), hence needing to observe it. We can't use `observeColumns` as it's not a column in the database, and this way seems to work 👀

I could be wrong though - I think @enahum or @avinashlng1080 have a lot more watermelondb experience and can chime in.

Reply:
That makes sense, but yes, let's wait for them to chime in before resolving the conversation.

Reply:
This is my understanding - 

1. By having a @lazy query, you would only get its result after you have intentionally requested it.  It is not loaded by default and hence, it provides a performance boast by loading only the records you need.

2. We have the `grand-parent`  **Categories** which holds all the `categories`  for the `currentTeamId`.  For each of its category, it renders **Category** component which is made up of `CategoryHeader` and `CategoryBody`.  Only the `CategoryHeader` needs to know if the current category hasChannels and thus it makes sense to trigger that lazy query at this level; where it is actually needed and used.